stac_generator/generator.py

```diff
@@ -34,32 +37,36 @@ def validate_data(self) -> bool:
 
     @abstractmethod
     def generate_item(self, location: str, counter: int) -> pystac.Item:
-        """Generate a STAC item from the provided file."""
+        """Generate a Stac item from the provided file."""
         raise NotImplementedError
 
-    @abstractmethod
     def write_items_to_api(self) -> None:
-        raise NotImplementedError
+        if self.items and self.collection:
+            api_items_url = f"{self.base_url}/collections/{self.collection.id}/items"
+            for item in self.items:
+                requests.post(api_items_url, json=item.to_dict())
 
     @abstractmethod
     def generate_catalog(self) -> pystac.Catalog:
-        """Generate a STAC catalog for the provided metadata implementation."""
+        """Generate a Stac catalog for the provided metadata implementation."""
         raise NotImplementedError
 
     @abstractmethod
     def generate_collection(self) -> pystac.Collection:
-        """Generate a STAC collection for the provided metadata implementation."""
+        """Generate a Stac collection for the provided metadata implementation."""
         raise NotImplementedError
 
-    @abstractmethod
     def write_collection_to_api(self) -> None:
-        raise NotImplementedError
+        # TODO: Build URL from components rather than hardcode here.
+        api_collections_url = f"{self.base_url}/collections"
+        if self.collection:
+            requests.post(api_collections_url, json=self.collection.to_dict())
 
-    @abstractmethod
     def write_to_api(self) -> None:
-        raise NotImplementedError
+        self.write_collection_to_api()
+        self.write_items_to_api()
 
-    @abstractmethod
     def validate_stac(self) -> bool:
-        """Check that the generated STAC is valid."""
-        raise NotImplementedError
+        if self.catalog and not self.catalog.validate():
+            return False
+        return not (self.collection and not self.collection.validate())
```

Comment:
I would prefer not to refactor the boolean logic like this. While not particularly difficult to parse, it is more difficult than what was there before.

Reply:
Same as [here](https://github.com/aus-plant-phenomics-network/stac-generator/pull/7#discussion_r1792835754)

Reply:
The linter did this?! What a jerk